### ISSUE TITLE
non-trainable variables update in training

### ISSUE THREAD

**Issue by skydoorkai** [sql-machine-learning/elasticdl#295]
Some keras layers have non-trainable variables, such as BatchNorm and RNN. These variables are not updated by gradients, but by update ops.

We need to run the update ops to update these internal states of the layers.
